doc/sphinx/source/dut-setup.txt

```diff
@@ -77,19 +80,14 @@ Connect using UART
 #. Set terminal options for the DUT serial connection by
    running ``stty`` command with appropriate settings::
 
-      # stty -F /dev/ttyS1 115200 cs8 -cstopb -parenb
-
-   .. note::
-
-      You need to adjust setting in the command above to fit your UART
-      connection
+      # stty -F /dev/ttyS2 115200 cs8 -cstopb -parenb
```

Comment:
Just so that i have it clear, are u saying that /dev/ttyS1 was your actual DUT? While testing this, the only serial console i could get was on /dev/ttyS2, i thought that was the only one that you can open.

Reply:
DUT means Device Under Test.

Let's say you're testing Tizen on a Raspberry Pi. You connect RPi's power to [MuxPi's power relay](https://wiki.tizen.org/MuxPi#DUT_Power) and RPi's serial to [MuxPi's DUT UART](https://wiki.tizen.org/MuxPi#UART). You use stm program to control the power relay to power on RPi and then you use ttyS1 to talk to RPi's serial.

Reply:
I see. I have tried to get a serial communication with the RPI but so far its been unsuccessful. The issue i have is that i cant seem to get the SD adapter to work. 

Although i can manage to give power to RPI via the power relay, i cant seem to get an image in it so that it boots. Meaning that the RPI never boots on to communicate on Serial. 

What step am I missing?

Reply:
A common problem with RPi is that the SD adapter cable is too long. I don't remember the reasons, maybe @amalinowsk2 does.

Reply:
I'm using a cable which is around 10-15cm long. Is there some other way to boot rpi? Can i use a USB cable for it? Where do i specify the change?

Reply:
I'm afraid it wont work unless you use some older /boot partition images. RPI3 firmware had 2 changes in last 2 years (@amalinowsk2 should be able to write more about this), i think it changed SD card reader frequencies which broke MuPi sd-mux part. After first change we had to shorten the cables to 1-3cm, after second one we had to use separate device - [SDWire](https://wiki.tizen.org/SDWire)

Reply:
It's true but this applies only (I think) to Tizen OS. I've tried several "normal" operating systems for RPI3 and all of them worked fine with 5cm long cables. But none of them worked wit 10cm cable. Tizen has ancient u-boot which could not be patched after changes of RPI3 firmware.
Anyway, don't use cables longer than 5cm for RPI3.

Reply:
> After first change we had to shorten the cables to 1-3cm 
This seems very unpractical. 
I tried to read the SD card adapter on my machine, but still no luck, can this stand true for other devices beside RPI3?

Reply:
Inconvenient but still usable. Remember that SD cards run at very high clock speed. Don't expect it to work with long cables.

Till now I know about only two boards that have some problems with MuxPi:
1. RPI3 after changes to its firmware - RPI3 firmware is not opened so we can't see what is the real problem but AFAIK there were some changes to VPU and SD clocks. I wouldn't blame MuxPi for that. After changes to the firmware there were reports that RPI3 stopped working even with some cards inserted directly into RPI3 slot.
2. HiKey - it was a nightmare, we had big problems with booting this device even with cards inserted directly into its slots.

As I said earlier, 5cm works fine with images like Raspbian, Snappy Ubuntu Core or Libre Elec.
Connect to RPI3 UART console and boot it with card inserted to it directly.
Then do the same with card connected through MuxPi and try to find where it fails.
If you get some I/O errors then you have to shorten the cable.

Reply:
> btw @fhajredini did you enable both uarts in armbian-config? One of them is used for communication between stm/muxpictl on nanopi and firmware on stm microcontroller on muxpi board. the other one is for communicating with DUT.

Yes, i have enabled both of them. Just now i enabled the third one as well. 

>Connect to RPI3 UART console and boot it with card inserted to it directly.
Then do the same with card connected through MuxPi and try to find where it fails.
If you get some I/O errors then you have to shorten the cable.

This sounds like a good way to debug it. Will do this.

Reply:
I changed it back to /dev/ttyS1. However i think its important to note that i never managed to get a serial connection from muxpi to rpi. I tried every serial port apart from ttyS1 and the only serial connection i could get was on stm (ttyS2). I solved this with using an arduino converter from UART to USB. I can now talk to the dut via /dev/ttyACM0. Still got UART on RPI tho.